src/app/catalog/recommendation/services/viewport.service.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2022 Carl Zeiss AG
+ * SPDX-License-Identifier: MIT
+ */
+
+import { ElementRef, Injectable } from '@angular/core';
+import { Observable } from 'rxjs';
+
+const options = {
+  root: null, // use the viewport
+  rootMargin: '0px',
+  threshold: 1.0, // trigger when 100% of the target is visible
+};
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ViewportService {
+  public isInViewport(elementRef: ElementRef<any>): Observable<boolean> {
+    return new Observable((observer) => {
+      const element = elementRef?.nativeElement.getElementsByTagName('p')[0];
```

Comment:
I would suggest that this element is provided as an argument instead of doing here the query (:

Reply:
good idea, fixed